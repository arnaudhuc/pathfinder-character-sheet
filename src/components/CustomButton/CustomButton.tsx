/** @jsx jsx */
import { jsx } from '@emotion/core';
import Button from '@material-ui/core/Button';

interface IButton {
	value: string;
	handleClick: () => void;
}

export const CustomButton: React.FC<IButton> = (props) => {
	const { value, handleClick } = props;
	return (
		<div>
			<Button onClick={handleClick} variant="contained" color="primary">
				{value}
			</Button>
		</div>
	);
};
