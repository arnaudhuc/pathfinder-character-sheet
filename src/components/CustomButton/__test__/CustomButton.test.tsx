/** @jsx jsx */
import { jsx } from '@emotion/core';
import renderer from 'react-test-renderer';
import { CustomButton } from '../CustomButton';

describe('CustomButton', () => {
	test('should render with default props', () => {
		const container = renderer
			.create(<CustomButton value="Next" handleClick={() => true} />)
			.toJSON();
		expect(container).toMatchSnapshot();
	});
});
