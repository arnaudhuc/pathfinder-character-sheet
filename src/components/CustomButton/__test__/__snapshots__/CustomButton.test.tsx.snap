// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton should render with default props 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Next
    </span>
  </button>
</div>
`;
